db/db_impl_write.cc

```diff
@@ -62,6 +62,11 @@ Status DBImpl::WriteImpl(const WriteOptions& write_options,
                          WriteBatch* my_batch, WriteCallback* callback,
                          uint64_t* log_used, uint64_t log_ref,
                          bool disable_memtable) {
+  // The current implementation does not support sync with concurrenet writes
+  assert(!concurrent_writes_ || !write_options.sync);
+  if (concurrent_writes_ && write_options.sync) {
```

Comment:
sure. I am trying to avoid 2pc specific names here. I will change it to concurrent_wal_writes_.